### PR TITLE
Implement interface of various inventories

### PR DESCRIPTION
**Pull request by FloofyPlasma**
# Description

Implements the interface of various inventories. It doesn't give them full functionality. **Please do not merge until TODO is complete!**

### TODO:
- [x] Generic_9x1
- [x] Generic_9x2
- [x] Generic_9x4
- [x] Generic_9x5
- [x] Generic_9x6 (Large chests)
- [x] Generic_3x3 (Dispenser & Dropper)
- [x] Anvil
- [x] Beacon
- [x] Blast Furnace
- [ ] Brewing Stand
- [ ] Enchantment Table ~~(Needs some type of currently unhandled packet, 0x15 otherwise disconnects.)~~
- [x] Furnace
- [ ] Grindstone
- [ ] Hopper
- [ ] Loom
- [ ] Merchant (Villager & Wandering trader)
- [ ] Shulker box
- [ ] Smithing table
- [x] Smoker
- [ ] Cartography table
- [ ] Stonecutter

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation comments
- [x] My changes generate no new warnings

### CODE REVIEW

File: Sources/Core/Sources/GUI/WindowType.swift [hunk 1]
```diff
@@ -121,7 +121,37 @@ public struct WindowType {
...
 
  // Generic window types
- public static let generic9x4 = WindowType(
+public static let generic9x1 = WindowType(
+    id: .vanilla(0),
+    identifier: Identifier(namespace: "minecraft", name: "generic_9x1"),
+    background: GUIElement.list(spacing: 0) {
+            GUIElement.sprite(.generic9x1)
+            GUIElement.sprite(.singleChestBottomHalf)
+        },
```
Comment (stackotter) [inline Sources/Core/Sources/GUI/WindowType.swift]: Remove the extra indent on lines 128-130

File: Sources/Core/Sources/GUI/GUISprite.swift [hunk 2]
```diff
@@ -22,6 +22,7 @@ public enum GUISprite {
   case singleChestTopHalf
   case singleChestBottomHalf
   case doubleChest // No top/bottom half since it just uses the full image
+  case generic9x1
   case generic9x4
   case generic9x5
```
Comment (stackotter) [inline Sources/Core/Sources/GUI/GUISprite.swift]: Add `TopHalf` to the end to make it clear that it's not the whole thing

File: Sources/Core/Sources/GUI/GUISprite.swift [hunk 3]
```diff
@@ -22,6 +22,7 @@ public enum GUISprite {
   case singleChestTopHalf
   case singleChestBottomHalf
   case doubleChest // No top/bottom half since it just uses the full image
```
Comment (stackotter) [inline Sources/Core/Sources/GUI/GUISprite.swift]: Feel free to rename these with their generic names instead. (also don't need the comment once the comment below is resolved cause it'll be clear which have halves and which don't).

**Review by stackotter — CHANGES_REQUESTED** [review]
Just left a few quick comments so that you can address them while you're working instead of having to go through a few cycles once the PR's done 😅 just formatting and naming stuff, some of which you already mentioned

**Comment by FloofyPlasma** [comment]
https://github.com/stackotter/delta-client/pull/196/commits/04b05a6dbe8f280194bb5dbea6f391b466bd1d0f implements dropper and dispenser, not beacon. I'm too tired, oops 😅

**Review by stackotter — APPROVED** [review]
Lgtm 🎉

**Pull request merged**